plugins/kubernetes/app/views/kubernetes_dashboard/_deploy_progress_widget.html.erb

```diff
@@ -10,17 +10,19 @@
         An error occurred while deploying {{buildLabel(lastRelease())}}.{{releaseId(lastRelease())}}.
       </span>
     </div>
-    <div class="deploy-group-release" ng-class="{ 'current-release': currentRelease(release), 'target-reached': targetStateReached(release) }"
-         ng-repeat="release in deployGroup.releases | orderBy:'id':true" ng-if="!deployCompleted()">
-      <div class="progress-wrapper">
-        <label>{{buildLabel(release)}}.{{releaseId(release)}}</label>
+    <div style="flex: auto;" ng-if="!deployCompleted() && !deployFailed()">
+      <div class="deploy-group-release" ng-class="{ 'current-release': currentRelease(release), 'target-reached': targetStateReached(release) }"
+           ng-repeat="release in deployGroup.releases | orderBy:'id':true">
+        <div class="progress-wrapper">
+          <label>{{buildLabel(release)}}.{{releaseId(release)}}</label>
```

Comment:
Initially I was using the objects directly and querying their properties, but after having to change the implementation a few times and having to to search across the html by all the references to the method properties in order to replace it, I switched over to methods.

By extracting it to a method, we can reuse it and if we ever need to change the implementation, we can just change it in the method and it will be applied everywhere it is used. Even if we need to rename the method to reflect what it is returning now, a refactor should do it, it will change the JS code but also the HTML code, as far as I remember.

So, in the end, a matter of preference, I would say, but I can say I got a few benefits from it. ;)